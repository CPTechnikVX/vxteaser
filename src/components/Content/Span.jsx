import React         from 'react';
import PropTypes     from 'prop-types';
import classnames    from 'classnames';
import CommonWrapper from '../CommonWrapper';

/**
 * Default span element
 */
export default class Span extends React.PureComponent {
	render() {
		const {children, modifier} = this.props;
		const classList            = [];

		if (modifier) {
			classList.push(modifier);
		}

		return <CommonWrapper childProps={this.props}><span className={classnames(classList)}>{children}</span></CommonWrapper>;
	}
}

Span.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	modifier: PropTypes.string,
};
