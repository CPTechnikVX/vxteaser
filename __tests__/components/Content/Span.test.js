import React    from 'react';
import {render} from '@testing-library/react';
import Span     from '../../../src/components/Content/Span';

describe('render common span', () => {
    test('with content', () => {
        const {container} = render(
                <Span top="10%" left="20%" modifier="testclass">This is the text!</Span>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
