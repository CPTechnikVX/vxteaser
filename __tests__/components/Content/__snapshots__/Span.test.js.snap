// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render common span with content 1`] = `
<span
  class="testclass"
  style="position: absolute; top: 10%; left: 20%;"
>
  This is the text!
</span>
`;
